gk/main.c

```diff
@@ -1493,16 +1532,31 @@ process_pkts_front(uint16_t port_front, uint16_t port_back,
 		case GK_REQUEST:
 			ret = gk_process_request(fe, &packet,
 				gk_conf->sol_conf);
+			if (ret >= 0) {
+				instance->pkts_num_request++;
+				instance->pkts_size_request +=
+					rte_pktmbuf_pkt_len(pkt);
+			}
 			break;
 
 		case GK_GRANTED:
 			ret = gk_process_granted(fe, &packet,
 				gk_conf->sol_conf);
+			if (ret >= 0) {
+				instance->pkts_num_granted++;
+				instance->pkts_size_granted +=
+					rte_pktmbuf_pkt_len(pkt);
+			}
 			break;
 
 		case GK_DECLINED:
 			ret = gk_process_declined(fe, &packet,
 				gk_conf->sol_conf);
+			if (ret >= 0) {
+				instance->pkts_num_declined++;
+				instance->pkts_size_declined +=
+					rte_pktmbuf_pkt_len(pkt);
```

Comment:
I think a packet in a declined flow will still have length `pkt_len` since it will not be encapsulated.